Packs/Base/Scripts/DBotFindSimilarIncidentsByIndicators/DBotFindSimilarIncidentsByIndicators.py

```diff
@@ -56,574 +61,469 @@ class FrequencyIndicators(BaseEstimator, TransformerMixin):
-
-
-if __name__ in ['__main__', '__builtin__', 'builtins']:
+    if incidents.empty:
+        return incidents
+
+    incident_ids = incidents.id.tolist() if INCIDENT_ID_FIELD in incidents.columns else incidents.index
+    res = execute_command(
+        "GetIncidentsByQuery",
+        args={
+            "query": f"incident.id:({' '.join(incident_ids)})",
+            "populateFields": ",".join(fields_to_display),
+        },
+        fail_on_error=True,
+    )
+    incidents_map: dict[str, dict] = {incident[INCIDENT_ID_FIELD]: incident for incident in json.loads(res)}
+    if CREATED_FIELD in fields_to_display:
+        incidents[CREATED_FIELD] = [
+            dateparser.parse(incidents_map[inc_id][CREATED_FIELD]).strftime(DATE_FORMAT)  # type: ignore
+            for inc_id in incident_ids
+        ]
+    if STATUS_FIELD in fields_to_display:
+        incidents[STATUS_FIELD] = [
+            STATUS_DICT.get(incidents_map[inc_id][STATUS_FIELD]) or " "
+            for inc_id in incident_ids
+        ]
+
+    for field in fields_to_display:
+        if field not in [CREATED_FIELD, STATUS_FIELD]:
+            incidents[field] = [incidents_map[inc_id][field] or "" for inc_id in incident_ids]
+    return incidents
+
+
+def replace_indicator_ids_with_values(
+    inc_ids: str,
+    indicators_data: dict[str, dict],
+) -> str:
+    return "\n".join([
+        indicators_data.get(x, {}).get(VALUE_FIELD) or " "
+        for x in inc_ids.split(" ")
+    ])
+
+
+def format_similar_incidents(
+    similar_incidents: pd.DataFrame,
+    indicators_data: dict[str, dict],
+    fields_to_display: list[str],
+) -> pd.DataFrame:
+    """ Formats the similar incidents DataFrame.
+
+    :param indicators_data: a mapping between IDs and the mutual indicators data
+    :param fields_to_display: Fields selected for enrichement
+    :return: a formatted, enriched DataFrame of the similar incidents.
+    """
+    if similar_incidents.empty:
+        return similar_incidents
+
+    # format and enrich DataFrame
+    similar_incidents = similar_incidents.reset_index().rename(columns={"index": INCIDENT_ID_FIELD})
+    similar_incidents[INCIDENT_LINK_COLUMN] = similar_incidents[INCIDENT_ID_FIELD].apply(
+        lambda _id: INCIDENT_LINK_FORMAT.format(_id)
+    )
+    similar_incidents[IDENTICAL_INDICATORS_COLUMN] = similar_incidents[IDENTICAL_INDICATORS_COLUMN].apply(
+        lambda inc_ids: replace_indicator_ids_with_values(inc_ids, indicators_data)
+    )
+    similar_incidents = similar_incidents[
+        [INCIDENT_LINK_COLUMN, INCIDENT_ID_FIELD, IDENTICAL_INDICATORS_COLUMN, SIMILARITY_SCORE_COLUMN]
+    ]
+    return enrich_incidents(similar_incidents, fields_to_display)
+
+
+def similar_incidents_results(
+    similar_incidents: pd.DataFrame,
+    indicators_of_actual_incident: dict,
+    fields_to_display: list,
+):
+    similar_incidents = format_similar_incidents(similar_incidents, indicators_of_actual_incident, fields_to_display)
```

Comment:
You an check whether `similar_incidents` is empty before calling `format_similar_incidents`.

Reply:
I do it inside the function